### PR TITLE
chore(web): blog overhaul

### PR DESCRIPTION
**Pull request by icepuma**
- make blog work again -> use new content layer stuff
- prepare blog series
- make sure drafts are not shown in featured list

### CODE REVIEW

**Comment by rawkode** [comment]
I can't run this locally:

```
Cannot destructure property 'type' of 'lookupMap[collection]' as it is undefined.
  Location:
    /home/rawkode/Code/src/github.com/RawkodeAcademy/RawkodeAcademy/projects/rawkode.academy/platform/web/node_modules/astro/dist/content/runtime.js:487:17
  Stack trace:
    at Object.eval [as transform] (/home/rawkode/Code/src/github.com/RawkodeAcademy/RawkodeAcademy/projects/rawkode.academy/platform/web/node_modules/astro/dist/content/runtime.js:487:17)
    at async ZodObject.safeParseAsync (file:///home/rawkode/Code/src/github.com/RawkodeAcademy/RawkodeAcademy/projects/rawkode.academy/platform/web/node_modules/zod/lib/index.mjs:731:24)
    at async parseData (file:///home/rawkode/Code/src/github.com/RawkodeAcademy/RawkodeAcademy/projects/rawkode.academy/platform/web/node_modules/astro/dist/content/content-layer.js:148:40)
    at async eval (/home/rawkode/Code/src/github.com/RawkodeAcademy/RawkodeAcademy/projects/rawkode.academy/platform/web/node_modules/astro/dist/content/loaders/glob.js:192:13)
```

**Comment by icepuma** [comment]
I fixed the `bun.lockb` file. I removed `.astro`, `dist`, `node_modules` and re-ran `bun install` and `ZITADEL_URL="aaa" ZITADEL_CLIENT_ID="bbb" bun run build` which works flawlessly :)

File: projects/rawkode.academy/platform/web/.npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+registry=https://registry.npmjs.org
```
Comment (rawkode) [inline projects/rawkode.academy/platform/web/.npmrc:1]: Why do we need this?

File: projects/rawkode.academy/platform/web/package.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "@astrojs/react": "4.0.0",
     "@astrojs/sitemap": "3.2.1",
     "@astrojs/tailwind": "5.1.3",
-    "@astrojs/ts-plugin": "^1.10.4",
+    "@astrojs/ts-plugin": "1.10.4",
```
Comment (rawkode) [inline projects/rawkode.academy/platform/web/package.json:21]: Do we need to pin this to a specific patch release?

File: projects/rawkode.academy/platform/web/src/content.config.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { defineCollection, reference, z } from "astro:content";
+
+const blog = defineCollection({
+  loader: glob({
+    pattern: ["**\/*.mdx"],
+    base: "./src/content/blog",
+  }),
+  schema: z.object({
+    title: z.string(),
+    series: reference("series").optional(),
```
Comment (rawkode) [inline projects/rawkode.academy/platform/web/src/content.config.ts:11]: This is cool

File: projects/rawkode.academy/platform/web/.npmrc (hunk 1 above)
Comment (icepuma) [reply projects/rawkode.academy/platform/web/.npmrc:1]: Yes! When you have a system-wide config which overrides this, the content of `bun.lockb` changes

File: projects/rawkode.academy/platform/web/package.json (hunk 2 above)
Comment (icepuma) [reply projects/rawkode.academy/platform/web/package.json:21]: This should not be a problem. When you run `bunx npm-check-updates --interactive` you'll get all the updates :)

File: projects/rawkode.academy/platform/web/.npmrc (hunk 1 above)
Comment (rawkode) [reply projects/rawkode.academy/platform/web/.npmrc:1]: Did you have a system wide config? 😛

**Review by rawkode — APPROVED** [review]

**Pull request merged**